plugins/modules/zypper.py

```diff
@@ -350,9 +363,21 @@ def parse_zypper_xml(m, cmd, fail_not_found=True, packages=None):
             return parse_zypper_xml(m, cmd, fail_not_found=fail_not_found, packages=packages)
 
         return packages, rc, stdout, stderr
+
+    if m.params['simple_errors']:
+        stdout = get_simple_errors(dom)
+
     m.fail_json(msg='Zypper run command failed with return code %s.' % rc, rc=rc, stdout=stdout, stderr=stderr, cmd=cmd)
 
 
+def get_simple_errors(dom):
+    simple_errors = []
+    message_xml_tags = dom.getElementsByTagName('message')
```

Comment:
In case `message_xml_tags` does not exist (it's probably `None` then?) you need to do something other than the `for` loop. How about returning `None`, and changing the invocation so that in case it returns `None`, it doesn't store the result in `stdout`?

Reply:
Added the change as per my understanding.  Let me know if that works.